### ISSUE TITLE
@types/aws-lambda: "Callback" is incorrectly typed

### ISSUE THREAD

**Issue by jasonzoladz** [DefinitelyTyped/DefinitelyTyped#16773]
@darbio, @skarum

First, thanks so much for putting these definitions together.  Very useful stuff.

This [line](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/index.d.ts#L246) which reads:

```javascript
export type Callback = (error?: Error, result?: any) => void;
```

should be:

```javascript
export type Callback = (error: Error | null, result: any) => void;
```
as the AWS docs read:

"**Parameters**:
- err (Error) — the error object returned from the request. Set to null if the request is successful.
- data (Object) — the de-serialized data returned from the request. Set to null if a request error occurs."

`error?: Error` means `Error | undefined`.  And the TypeScript docs state: 

"`In TypeScript, both undefined and null actually have their own types named undefined and null respectively.`"

**Comment by darbio** [comment]
Good find @jasonzoladz - I hadn't read that  it if the doco, and just assumed it would be null. Thanks for the heads up! Do you want to pull request it?

**Comment by jasonzoladz** [comment]
It's perhaps a little small for a PR.  :)

However, I do plan to drop a PR in the near term to add additional event types.

**Comment by darbio** [comment]
Great! We'd just be PRing it too, so if you roll it all up into one when you do the PR that would save everyone some time 👍🏼

**Comment by OrthoDex** [comment]
Hey @darbio 
I ran into this issue as well and I had to do a quick hack to bypass it. Do you mind if I made a quick PR?

**Comment by darbio** [comment]
Go for it! Happy to review.

**Comment by weswigham** [comment]
Fixed in #20381
